farm/infer.py

```diff
@@ -54,9 +56,28 @@ def __init__(self, load_dir, batch_size=4, gpu=False):
             self.label_map = self.processor.label_maps[0]
         elif len(self.model.prediction_heads) == 0:
             self.prediction_type = "embedder"
-        self.name = os.path.basename(load_dir)
+        # self.name = os.path.basename(load_dir)
```

Comment:
Suggestion for the name: 
- We make it an optional argument in the constructor
- In case of loading from dir, we still pass the directory name to it
- In case of initializing from building blocks, users can supply a name if they want
- If no name is supplied: the name is set to `self.name = f"anonymous-{self.prediction_type}"`